reyna/src/com/b2msolutions/reyna/services/StoreService.java

```diff
@@ -123,7 +140,11 @@ private void insert(Message message) {
                 this.repository.insert(message, getStorageSizeLimit(this));
             }
 
-            this.startService(new Intent(this, ForwardService.class));
+            if (!this.preferences.getBatchUpload()) {
```

Comment:
I am trying to minimize the impact of launching an a service that most of the time will exit immediately. most of the time there is no enough data.

better to leave this to the alarm or networkchange actions

@Youhana-Hana